web/src/main/webapp/js/graphhopper/GHRequest.js

```diff
@@ -159,7 +163,7 @@ GHRequest.prototype.createGPXURL = function (withRoute, withTrack, withWayPoints
 };
 
 GHRequest.prototype.createHistoryURL = function () {
-    return this.createPath("?" + this.createPointParams(true));
+    return this.createPath("?" + this.createPointParams(true)) + "&useMiles=" + !!this.useMiles;
```

Comment:
Ah, ok.

> while use_miles (and do_zoom (?)) are more about the client side representation of the results.

Yes, we are storing state in the URL. Although a bit ugly, kind of common.